imap/ical_support.h

```diff
@@ -187,6 +187,8 @@ extern void icaltimezone_truncate_vtimezone_advanced(icalcomponent *vtz,
                                                      icaltimetype *last_dtstart,
                                                      int ms_compatible);
 
+extern int ical_categories_is_color(icalproperty *cprop);
```

Comment:
What is the purpose of `extern` here?

Reply:
The function `ical_categories_is_color()` is only declared here (or anywhere that includes this file), its definition is elsewhere (i.e. in imap/ical_support.c).  It's usual to use the `extern` keyword like this in header files.

Historically, `extern` was also the default, so old code often omits it, but it's good practice to include it.

Reply:
`extern` is extra text to read, which does nothing.  It is obvious, that the definition is not on the line after the prototype.

Extra code, which does nothing, complicates understanding of the code.  Other examples of extra code, which does nothing, are defined, but unused functions.